### PR TITLE
Provide global variables to Console through Context API

### PR DESCRIPTION
**Pull request by thiva-k**
### Purpose
Currently global variables `isAdaptiveAuthenticationAvailable` and `isOrganizationManagementEnabled` are declared in the index.html and hoisted as global variables to be used within features package. This prevents reusing of features package since, it is dependent on this external global variables. This PR changes this implementation and passes these global variables through context within the features package itself, preventing any external dependencies for these variables. Further, the user of this package can override these variables if needed.

### Related Issues
- product-is: https://github.com/wso2/product-is/issues/20119

### Related PRs
- identity-apps: https://github.com/wso2/identity-apps/pull/5849

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/index.tsx [hunk 1]
```diff
@@ -60,27 +61,29 @@ const RootWithConfig = (): ReactElement => {
...
-                                        <ProtectedApp />
-                                    </OrganizationsProvider>
-                                </AppConfigProvider>
-                            </AuthProvider>
-                        </BrowserRouter>
-                    </UserPreferencesProvider>
-                </Provider>
-            </ThemeProvider>
-        </AppSettingsProvider>
+        <GlobalVariablesProvider >
```
[the remark this answers is not in the document]
Comment (thiva-k) [reply apps/console/src/index.tsx]: Added

File: features/admin.core.v1/hooks/use-global-variables.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useContext } from "react";
+import GlobalVariablesContext, { GlobalContextType } from "../context/global-variables-context";
+
+const useGlobalVariables = (): GlobalContextType => {
```
Comment (brionmario) [inline features/admin.core.v1/hooks/use-global-variables.ts:22]: Add a TSDoc.

File: features/admin.core.v1/providers/global-variables-provider.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import React from "react";
+import GlobalVariablesContext, { GlobalContextType, defaultValues } from "../context/global-variables-context";
+
+interface GlobalProviderProps {
+  value?: Partial<GlobalContextType>;
+  children: React.ReactNode;
+}
+
+const GlobalVariablesProvider: React.FC<GlobalProviderProps> = ({ children, value = {} }: GlobalProviderProps) => {
```
Comment (brionmario) [inline features/admin.core.v1/providers/global-variables-provider.tsx:27]: Add a TSDoc comment.

File: features/admin.core.v1/context/global-variables-context.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Context, createContext } from "react";
+
+export interface GlobalContextType {
+  isAdaptiveAuthenticationAvailable: boolean;
```
[the remark this answers is not in the document]
Comment (thiva-k) [reply features/admin.core.v1/context/global-variables-context.tsx:26]: Added

File: features/admin.core.v1/hooks/use-global-variables.ts (hunk 2 above)
Comment (thiva-k) [reply features/admin.core.v1/hooks/use-global-variables.ts]: Added

File: features/admin.core.v1/providers/global-variables-provider.tsx (hunk 3 above)
Comment (thiva-k) [reply features/admin.core.v1/providers/global-variables-provider.tsx:39]: Added

**Review by JayaShakthi97 — DISMISSED** [review]

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**